realtime_connext/poll_utils.cxx

```diff
@@ -0,0 +1,468 @@
+     * code and comment out the create_datareader call above. */
+
+    /*
+    DDS_DataReaderQos datareader_qos;
+    retcode = subscriber->get_default_datareader_qos(datareader_qos);
+    if (retcode != DDS_RETCODE_OK) {
+        printf("get_default_datareader_qos error\n");
+        return -1;
+    }
+
+    datareader_qos.history.kind = DDS_KEEP_ALL_HISTORY_QOS;
+
+    reader = subscriber->create_datareader(
+        topic, datareader_qos, NULL,
+        DDS_STATUS_MASK_ALL);
+    if (reader == NULL) {
+        printf("create_datareader error\n");
+        subscriber_shutdown(participant);
+        return -1;
+    }
+    */
+
+    pollDataReader *poll_reader = pollDataReader::narrow(reader);
+    if (poll_reader == NULL) {
+        printf("DataReader narrow error\n");
+        return -1;
+    }
+
+    if (rttest_set_sched_priority(97, SCHED_RR) != 0) {
+      perror("Failed to set scheduling priority");
+    }
+
+    if (rttest_lock_memory() != 0) {
+      perror("Couldn't lock memory");
+    }
+    rttest_prefault_stack();
+
+    /* Main loop */
+    rttest_spin(subscriber_callback, static_cast<void*>(poll_reader));
+
+    /* Delete all entities */
+    status = subscriber_shutdown(participant);
+
+    rttest_write_results_file("rttest_subscriber_results");
+    rttest_finish();
+
+    return status;
+}
+
+int subscriber_main(Arguments *args)
+{
+    return subscriber_main(args->argc, args->argv);
+}
+
+void *subscriber_main(void *args)
+{
+    Arguments *new_args = static_cast<Arguments*>(args);
+    int ret = subscriber_main(new_args);
+    return static_cast<void*>(&ret);
+}
+
+/* Delete all entities */
+static int publisher_shutdown(
+    DDSDomainParticipant *participant)
+{
+    DDS_ReturnCode_t retcode;
+    int status = 0;
+
+    if (participant != NULL) {
+        retcode = participant->delete_contained_entities();
+        if (retcode != DDS_RETCODE_OK) {
+            printf("delete_contained_entities error %d\n", retcode);
+            status = -1;
+        }
+
+        retcode = DDSTheParticipantFactory->delete_participant(participant);
+        if (retcode != DDS_RETCODE_OK) {
+            printf("delete_participant error %d\n", retcode);
+            status = -1;
+        }
+    }
+
+    /* RTI Connext provides finalize_instance() method on
+       domain participant factory for people who want to release memory used
+       by the participant factory. Uncomment the following block of code for
+       clean destruction of the singleton. */
+    retcode = DDSDomainParticipantFactory::finalize_instance();
+    if (retcode != DDS_RETCODE_OK) {
+        printf("finalize_instance error %d\n", retcode);
+        //status = -1;
+    }
+
+    return status;
+}
+
+void *publisher_callback(void *args)
+{
+    PublisherNode *pub_node = static_cast<PublisherNode*>(args);
+    // message instance, instance_handle, poll_writer
+    /* Set x to a random number between 0 and 9 */
+    pub_node->instance->seq = (int)(rand()/(RAND_MAX/10.0));
```

Comment:
This is a relic from the RTI example that I based this on:

https://github.com/rticommunity/rticonnextdds-examples/blob/master/examples/polling_read/c%2B%2B/poll_publisher.cxx

I can change it, since sequence numbers really aren't supposed to be random integers betwene 0 and 9...